slices/slices.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slices
+
+// Map returns a slice consisting of the results of applying the given function
+// to the elements of the given slice.
+func Map[T1, T2 any](slice []T1, mapper func(T1) T2) []T2 {
+	if mapper == nil {
+		panic("nil mapping function provided")
```

Comment:
It's hardly ever the case that code should panic: go/go-style/decisions#dont-panic. Returning an error and letting the caller handle it is more graceful.

Reply:
providing a `nil` mapping function is basically an egregious error. This is just attempting to provide a more informative message than the standard nil pointer deference. I don't think it worth polluting the API with an error worth it for this niche case.